### PR TITLE
Skip testPitCreatedOnReplica IT with remote store

### PR DESCRIPTION
**Pull request by dreamer-89**
### Description
Mute `testPitCreatedOnReplica` test causing a higher gradle check failures in recent runs. Based on issue description and linked PRs which failed, it seems it is failing mostly on 2.x branch.

### Related Issues
Related #https://github.com/opensearch-project/OpenSearch/issues/8850

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Poojita-Raj — APPROVED** [review]

File: server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java [hunk 1]
```diff
@@ -1177,6 +1177,10 @@ public void testScrollWithOngoingSegmentReplication() throws Exception {
     }
 
     public void testPitCreatedOnReplica() throws Exception {
+        assumeFalse(
+            "Skipping the test as its not compatible with segment replication with remote store yet.",
```
Comment (Poojita-Raj) [inline server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java:1181]: Could we add the issue number tracking this issue as well? Like how we do for awaitsFix?

Comment (kotwanikunal) [reply server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java:1181]: +1

Comment (dreamer-89) [reply server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java]: Updated.

**Review by kotwanikunal — APPROVED** [review]

**Comment by dreamer-89** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> * **RESULT:** FAILURE ❌
> * **URL:** https://build.ci.opensearch.org/job/gradle-check/20947/
> * **CommitID:** [72377d3](https://github.com/opensearch-project/OpenSearch/commit/72377d3bfaa546bab8f462f1e359f0c6aa9827d5)
>   Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>   Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

https://github.com/opensearch-project/OpenSearch/issues/5031

**Pull request merged**